mat/cholesky.go

```diff
@@ -1065,12 +1071,66 @@ func (c *PivotedCholesky) Rank() int {
 
 // Cond returns the condition number of the factorized matrix.
 func (c *PivotedCholesky) Cond() float64 {
-	if !c.ok {
+	if c.chol == nil {
 		panic(badCholesky)
 	}
 	return c.cond
 }
 
+// RawU returns the Triangular matrix used to store the Cholesky factorization
+// of the original matrix A. If the returned matrix is modified, the
+// factorization is invalid and should not be used.
+func (c *PivotedCholesky) RawU() Triangular {
```

Comment:
> This can return a Triangular((*TriDense)(nil)).

Do you mean when `c.chol == nil`?

> Probably we should note that the returned value is only usable if Factorize has been called — is that too much hand-holding?

I don't know if too much but I've added it to the comment.

Reply:
> Do you mean when `c.chol == nil`?

Yep

Reply:
Done.

Reply:
Sorry, this is not what I meant (English is slightly ambiguous here). The explicit `return Triangular((*TriDense)(nil))` is already the behaviour when `c.chol == nil` since `c.chol` is a `*TriDense`. My concern is that the caller can receive a  `Triangular((*TriDense)(nil))` which that can't easily check is valid. So we should either comment that the return is valid only if `Factorize` was called (unless we think that is too much handholding), or maybe return a literal nil which is easily checked.

Reply:
I did doubt but was distracted and misunderstood. I changed to return nil and added a comment. PTAL